### PR TITLE
Parameters for advanced item and label styling of tree nodes

### PR DESCRIPTION
**Pull request by fo-code**
I added additional parameters in order to be able to specifically style borders and labels of tree nodes.
These are useful for also styling outer tree nodes with children for example.

Initially, I used the default values from the echarts documentation if these parameters are not set specifically.
This might override level stylings for tree maps which are specified in Javascript files.
Maybe it would be an option to set these not specifically set values to `null` in order to avoid this, @uhafner ?
What do you prefer here?

### CODE REVIEW

File: src/main/java/edu/hm/hafner/echarts/UpperLabel.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Label style for an item with children.
+ * <p>
+ * This class will be automatically converted to a JSON object.
+ * </p>
+ *
+ * @author Florian Orendi
+ */
+public class UpperLabel {
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/echarts/UpperLabel.java:11]: Why is this a copy of the `Label` class? Can't we simply have a property `upperLabel` of the `Label` in class `TreeMapNode`?

File: src/test/java/edu/hm/hafner/echarts/TreeMapNodeTest.java [hunk 2]
```diff
@@ -20,6 +20,14 @@ void shouldCreateNode() {
         assertThat(root.getItemStyle()).satisfies(
                 itemStyle -> assertThat(itemStyle.getColor()).isEqualTo("-")
         );
+        assertThat(root.getLabel()).satisfies(
+                label -> assertThat(label.getShow()).isEqualTo(false),
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/echarts/TreeMapNodeTest.java:24]:
```suggestion
                label -> assertThat(label.getShow()).isFalse(),
```

**Comment by uhafner** [comment]
> Initially, I used the default values from the echarts documentation if these parameters are not set specifically.
This might override level stylings for tree maps which are specified in Javascript files.

Do we have those maps already? Currently, the code coverage plugin is the only user of these charts.

> Maybe it would be an option to set these not specifically set values to null in order to avoid this, @uhafner ?
What do you prefer here?

I think this is not required. If someone complains we can add that later.

**Review by uhafner — CHANGES_REQUESTED** [review]

**Comment by fo-code** [comment]
> Do we have those maps already? Currently, the code coverage plugin is the only user of these charts.

No, not within the coverage plugin. I just wanted to mention it because I did not know if this is in use in other plugins. :)

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
After locally installing the tree charts are now broken (no label). That means we need to release both changes in sync:

Bildschirmfoto 2022-09-21 um 13 49 39

**Pull request merged**